### PR TITLE
add mattermost entreprise edition

### PR DESCRIPTION
**Pull request by LiquidITGuy**
First of all, thank you for your contribution! 😄

### ☑️ Self Check before Merge

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [x] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [x] Icon is added as a png file to the logos directory.

### CODE REVIEW

**Comment by LiquidITGuy** [comment]
Got an issue with the docker image (
```
 Error: (HTTP code 404) unexpected - manifest for mattermost/mattermost-enterprise-edition:v5.31.1 not found: manifest unknown: manifest unknown
captain-captain.1.wfea24oyw33c@sandox    | Error: (HTTP code 404) unexpected - manifest for mattermost/mattermost-enterprise-edition:v5.31.1 not found: manifest unknown: manifest unknown
captain-captain.1.wfea24oyw33c@sandox    |     at /usr/src/app/node_modules/docker-modem/lib/modem.js:301:17
captain-captain.1.wfea24oyw33c@sandox    |     at IncomingMessage.<anonymous> (/usr/src/app/node_modules/docker-modem/lib/modem.js:328:9)
captain-captain.1.wfea24oyw33c@sandox    |     at IncomingMessage.emit (events.js:327:22)
captain-captain.1.wfea24oyw33c@sandox    |     at endReadableNT (_stream_readable.js:1327:12)
captain-captain.1.wfea24oyw33c@sandox    |     at processTicksAndRejections (internal/process/task_queues.js:80:21)
captain-captain.1.wfea24oyw33c@sandox    | POST /api/v2/user/apps/appData/aaa 500 1001.685 ms - 21
```
)

**Comment by kgnfth** [comment]
there is no v5.31.1 tag see here https://hub.docker.com/r/mattermost/mattermost-enterprise-edition/tags

**Comment by LiquidITGuy** [comment]
Thank you @kgnfth I'm feeling so stupid right now.
I got confused with the 2 services I tried to upgrade at the same time (jitsi and mattermost)
I will fix the PR tomorrow
Thank you so much :)

File: public/v4/apps/mattermost-ee.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            containerHttpPort: '8065'
+caproverOneClickApp:
+    variables:
+      - id: $$cap_pg_version
+        label: postgres Docker Image tag
+        defaultValue: 13.1
+        description: Check out their Docker page for the valid tags https://hub.docker.com/_/postgres
+      - id: $$cap_mattermost_version
+        label: mattermost Docker Image tag
+        defaultValue: v5.31.1
```
Comment (githubsaturn) [inline public/v4/apps/mattermost-ee.yml:40]: This is not a tag for this image: https://hub.docker.com/r/mattermost/mattermost-enterprise-edition/tags?page=1&ordering=last_updated

Did you mean to use mattermost-enterprise-edition or mattermost-prod-app?

Comment (LiquidITGuy) [reply public/v4/apps/mattermost-ee.yml:37]: thanks,
fixed

**Comment by githubsaturn** [comment]
Thanks

**Pull request merged**